// google-utils/index.js
const { Client } = require("@googlemaps/google-maps-services-js");

const client = new Client({});

const getNearbyPlaces = async (apiKey, location, radius = 2000, pageToken) => {
    const googleRes = (await client
        .placesNearby({
            params: {
                key: apiKey,
                location,
                radius,
                type: 'restaurant',
                keyword: 'restaurant',
                pagetoken: pageToken,
            }
        })).data;

    const { results,  next_page_token: nextPageToken } = googleRes;
    return { results,  nextPageToken };
}

const getPlacesAutocomplete = async (apiKey, location, radius, query, sessiontoken) => {
    const googleRes = (await client
        .placeAutocomplete({
            params: {
                key: apiKey,
                location,
                radius,
                origin: location,
                types: 'establishment',
                input: query,
                sessiontoken,
            }
        })).data;

    console.log("get places autocomplete googgle res = '", googleRes);

    const { predictions } = googleRes;
    return predictions;
}

module.exports = {
    getNearbyPlaces,
    getPlacesAutocomplete,
}
